zboss/Kconfig

```
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+menu "ZBOSS stack configuration."
```

This is a followup to comment: https://github.com/NordicPlayground/nrfxlib/pull/166/commits/e563e205dc5150a57198b7ba3172246cac1f6cfb#r420684509

Feel free to move the depends to other Kconfig, in case you find it more appropriate to be located elsewhere.

But looking at lib folder, I only find: 
`zboss/lib/cortex-m4/hard-float`
so I would assume something like:
```suggestion
menu "ZBOSS stack configuration."
        depends on CPU_CORTEX_M4 && FPU
```
will prevent the user from selecting Zboss if not using a Cortex-M4, or when using soft-float.

---

thanks for this suggestion,
I will narrow `depends on` to specific SoCs that are currently supported by zigbee: SOC_NRF52840 and (possibly) SOC_NRF52833.
I think Kconfig of zigbee subsys is a good place to add this dependency at `menuconfig ZIGBEE`. It will impact ZBOSS as well because all its options are enclosed with `if ZIGBEE`

Zigbee subsystem pull request:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2208

---

I coundn't find `FPU`. Did you mean `CPU_HAS_FPU`?

---

because I was looking upstream, the right name in our fork is `FLOAT`.
NCS Fork: `FLOAT`
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/arch/Kconfig#L509

Zephyr upstream: `FPU`
https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/Kconfig#L525